Update prom doc as default has changed (#7568)
istio/istio.io: content/en/docs/ops/integrations/prometheus/index.md

To simplify the configuration of metrics, Istio offers two modes of operation.

### Option 1: Metrics merging

To simplify configuration, Istio has the ability to control scraping entirely by `prometheus.io` annotations. This allows Istio scraping to work out of the box with standard configurations such as the ones provided by the [Helm `stable/prometheus`](https://github.com/helm/charts/tree/master/stable/prometheus) charts.

{{< tip >}}
While `prometheus.io` annotations are not a core part of Prometheus, they have become the de facto standard to configure scraping.
{{< /tip >}}

> This option is enabled by default, and can be configured by setting `--set meshConfig.enablePrometheusMerge=true|false` during [installation](/docs/setup/install/istioctl/). When this setting is enabled, appropriate `prometheus.io` annotations will be added to all workloads to set up scraping. If these annotations already exists, they will be overwritten. In these case, the Envoy sidecar will merge Istio's metrics with the application metrics.

Reviewer: What does enabled by default mean? In all configuration profiles, the default profile, demo profile?

Author: in all profiles

> This option is enabled by default, and can be configured by setting `--set meshConfig.enablePrometheusMerge=true|false` during [installation](/docs/setup/install/istioctl/). When this setting is enabled, appropriate `prometheus.io` annotations will be added to all workloads to set up scraping. If these annotations already exists, they will be overwritten. In these case, the Envoy sidecar will merge Istio's metrics with the application metrics.

Reviewer: ```suggestion
This option is enabled by default but can be disabled by passing `--set meshConfig.enablePrometheusMerge=false` during [installation](/docs/setup/install/istioctl/). When enabled, appropriate `prometheus.io` annotations will be added to all workloads to set up scraping. If these annotations already exist, they will be overwritten. With this option, the Envoy sidecar will merge Istio's metrics with the application metrics.
```
